src/compile/data/optimize.ts

```diff
@@ -185,11 +186,9 @@ function optimizationDataflowHelper(dataComponent: DataComponent) {
 
   mutatedFlag = roots.map(mergeParse).some(isTrue) || mutatedFlag;
   mutatedFlag = roots.map(optimizers.mergeIdenticalNodes).some(isTrue) || mutatedFlag;
-  keys(dataComponent.sources).forEach(s => {
-    if (dataComponent.sources[s].numChildren() === 0) {
-      delete dataComponent.sources[s];
-    }
-  });
+
+  dataComponent.sources = roots;
```

Comment:
Both are the same array.

Reply:
No, we use filter to remove things from the array and that is not in place.